src/infra/oracle/redStoneService.ts

```diff
@@ -0,0 +1,29 @@
+import {  ethers } from "ethers";
+import { Address } from "../../utils/constants";
+
+
+export class RedstoneOracleService {
+  private address;
+  private provider;
+
+  constructor(oracleAddress: Address, provider: ethers.Provider) {
+    this.address = oracleAddress;
+    this.provider = provider;
+  }
+
+  priceOf = async (asset: Address) => {
+    const oracleContract = new ethers.Contract(this.address, [
+      "function priceOf(address asset) external view returns (uint256)"
+    ], this.provider);
+
+    return await oracleContract.priceOf(asset);
```

Comment:
You should not await if you dont do anything after that, just return the promise :)